crates/milli/src/search/facet/facet_distribution.rs

```diff
@@ -287,37 +288,23 @@ impl<'a> FacetDistribution<'a> {
     }
 
     pub fn compute_stats(&self) -> Result<BTreeMap<String, (f64, f64)>> {
-        let fields_ids_map = self.index.fields_ids_map(self.rtxn)?;
-        let filterable_fields = self.index.filterable_fields(self.rtxn)?;
         let candidates = if let Some(candidates) = self.candidates.clone() {
             candidates
         } else {
             return Ok(Default::default());
         };
 
-        let fields = match &self.facets {
-            Some(facets) => {
-                let invalid_fields: HashSet<_> = facets
-                    .iter()
-                    .map(|(name, _)| name)
-                    .filter(|facet| !crate::is_faceted(facet, &filterable_fields))
-                    .collect();
-                if !invalid_fields.is_empty() {
-                    return Err(UserError::InvalidFacetsDistribution {
-                        invalid_facets_name: invalid_fields.into_iter().cloned().collect(),
-                        valid_facets_name: filterable_fields.into_iter().collect(),
-                    }
-                    .into());
-                } else {
-                    facets.iter().map(|(name, _)| name).cloned().collect()
-                }
-            }
-            None => filterable_fields,
-        };
+        let fields_ids_map = self.index.fields_ids_map(self.rtxn)?;
+        let fields_ids_map = FieldIdMapWithMetadata::new(
```

Comment:
spawning a field id map with metadata might be a perf regression here, to check with benchmarks

depending on the results it might be better to do this more manually by matching the filterable attributes rules only of the members of `facets`

this is only while we don't store the `FieldIdMapWithMetadata` in the index data